rdkPlugins/Storage/source/Storage.h

```diff
@@ -78,12 +80,123 @@ class Storage : public RdkPluginBase
     // persistent option is selected
     bool postStop() override;
 
+    // In this hook there should be deletion of img file when non-
+    // persistent option is selected
+    bool postHalt() override;
+
 public:
     std::vector<std::string> getDependencies() const override;
 
-    std::vector<LoopMountDetails::LoopMount> getLoopMounts();
-    std::vector<std::unique_ptr<LoopMountDetails>> getLoopDetails();
+private:
+    std::vector<MountProperties> getLoopMounts();
+    std::vector<MountProperties> getBindLoopMounts();
+    std::vector<std::unique_ptr<MountDetails>> getMountDetails();
 
+    template<typename T>
```

Comment:
My preference would be to have these functions in a cpp file, rather than the header if possible.

Reply:
I created template functions because I want to do the same thing with different unrelated parameter types:
- CreateMountProperties() can be called with `rt_defs_plugins_storage_data_bindloop_element*` and `rt_defs_plugins_storage_data_loopback_element*`;
- CreateMountDetails() can be called to create a LoopMountDetails or BindLoopMountDetails object depending on how you call it (e.g. `auto bindLoopMount = CreateMountDetails<BindLoopMountDetails>(properties);`)